src/server/api/routers/results.ts

```diff
@@ -249,96 +259,143 @@ async function generateImpactPayouts(round: Round, db: PrismaClient) {
     },
     {} as Record<string, number>,
   );
-
   console.log("metricAmounts", metricAmounts);
 
   // Fetch metrics from the CSV
   const projectMetrics = await fetchImpactMetricsFromCSV();
-
   console.log("projectMetrics", projectMetrics);
 
-  // Used to calculate the percentage contribution of each project towards each metric,
-  // which then helps in distributing the metricAmounts proportionally
+  // Used to calculate the total score for each metric across all projects
   const totalMetricScoresFromCSV = Object.keys(metricAmounts).reduce(
     (accumulator, key) => {
-      // Initialize the total for each metric key
       accumulator[key] = 0;
-
-      // Iterate over each project
       for (const projectMetric of projectMetrics) {
         if (key in projectMetric) {
-          // Add the project's value for this metric key to the total
           accumulator[key] += projectMetric[key as MetricId] as number;
         }
       }
-
       return accumulator;
     },
     {} as Record<string, number>,
   );
-
   console.log("totalMetricScoresFromCSV", totalMetricScoresFromCSV);
 
-  const projectPayouts: BallotResults = {};
-  let totalMetricsAmount = 0;
+  const projectPayouts: Record<string, BallotResults> = {};
+  const totalAmountForRound = getTotalAmountForImpactRound();
+
+  // Iterate over each award
+  for (const award of awards) {
```

Comment:
if we don't have any awards, can we put all projects and metrics into one awards by default to mimic the behavior we would have without awards?

Reply:
Yup ! That's exactly what i had in mind